service/src/main/kotlin/io/provenance/explorer/domain/entities/Transactions.kt

```diff
@@ -474,3 +478,66 @@ class TxNftJoinRecord(id: EntityID<Int>) : IntEntity(id) {
     var metadataId by TxNftJoinTable.metadataId
     var metadataUuid by TxNftJoinTable.metadataUuid
 }
+
+object TxEventsTable : IntIdTable(name = "tx_events") {
+    val blockHeight = integer("block_height")
+    val txHash = varchar("tx_hash", 64)
+    val txHashId = reference("tx_hash_id", TxCacheTable)
+    val txMessageId = text("tx_message_id")
+    val txMsgTypeId = varchar("tx_message_type_id", 128)
+    val eventType = varchar("event_type", 256)
+    val txMessageHash = text("tx_message_hash")
+}
+
+class TxEventRecord(id: EntityID<Int>) : IntEntity(id) {
+    companion object : IntEntityClass<TxEventRecord>(TxEventsTable) {
+
+        private fun findByTxHashAndMessageHash(txHashId: Int, messageHash: String) = transaction {
+            TxEventRecord.find { (TxEventsTable.txHashId eq txHashId) and (TxEventsTable.txMessageHash eq messageHash) }
+                .firstOrNull()
+        }
+
+        fun insert(blockHeight: Int, txHash: String, txId: EntityID<Int>, msg: Any, type: String, msgId: String, msgTypeId: String): String {
+            var eventId = ""
+            transaction {
+                findByTxHashAndMessageHash(txId.value, msg.value.toDbHash()) ?: TxEventsTable.insert {
+                    it[this.blockHeight] = blockHeight
+                    it[this.txHash] = txHash
+                    it[this.txHashId] = txId
+                    it[this.txMessageId] = msgId
+                    it[this.txMsgTypeId] = msgTypeId
+                    it[this.eventType] = type
+                    it[this.txMessageHash] = msg.value.toDbHash()
+                }
+                eventId = this.id
+            }
+            return eventId
+        }
+    }
+}
```

Comment:
You are missing the var properties in the record object. See https://github.com/provenance-io/explorer-service/blob/main/service/src/main/kotlin/io/provenance/explorer/domain/entities/Transactions.kt#L240-L297